packages/lib/contracts/upgradeability/UpgradeabilityProxy.sol

```diff
@@ -26,10 +26,15 @@ contract UpgradeabilityProxy is Proxy {
   /**
    * @dev Contract constructor.
    * @param _implementation Address of the initial implementation.
+   * @param _data Data to send as msg.data in the low level call.
+   * It should include the signature and the parameters of the function to be called, as described in
+   * https://solidity.readthedocs.io/en/develop/abi-spec.html#function-selector-and-argument-encoding.
    */
-  constructor(address _implementation) public {
+  constructor(address _implementation, bytes _data) public payable {
     assert(IMPLEMENTATION_SLOT == keccak256("org.zeppelinos.proxy.implementation"));
-
+    if(_data.length > 0) {
+      require(_implementation.delegatecall(_data));
```

Comment:
This means we are settling with `initializer functions` and not `initializer contracts`. Should we consider instead accepting `address implementation, address initializer, bytes data`, and run `initializer.delegatecall(data)` if initializer is not zero? The App could initially use the same address for both, but leaves the door open for initializer contracts in the future. @frangio care to chime in here?

Reply:
I'd like to leave the door open but I'm worried that it might be overkill for now. Would it be too disruptive to change that in a future version?

Note that if we do add the `address initializer` parameter, we should allow `_data.length == 0` because initializer contracts can run with no data.

Reply:
I'd like the idea of `initializer contracts` but in that case, I'd expect the CLI to solve the initializers functions to initializer contracts conversions and it might be an overkill for now. I think we can provide a new implementation of the proxy using `initializer contracts` with the required CLI improvements to deal with it in the future.

Reply:
So the CLI change would be transparent and backwards compatible. But the `Proxy` and `App` changes would not be backwards compatible and would need a major bump.

Are we comfortable with that?

Reply:
The App change could be backwards compatible, as we could just add a new `createAndCall` method that receives the `initializer address` parameter as well as the others. The only breaking change would be around proxies, but since it only affects the ctor, I don't think it'd be breaking at all.

Reply:
What do we mean by "breaking" exactly? If we do change the constructor it _could_ break deployment scripts that use the `Proxy` directly.